app/src/redux/data/dataSlice.ts

```diff
@@ -0,0 +1,198 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { FilmData } from '../../components/filmData';
+import fetchMovies from './movieHttp';
+
+export interface SortData {
+    title: string;
+    direction: string;
+    value: string;
+}
+
+interface DataSliceInterface {
+    loading: boolean;
+    error: string | undefined;
+    data: Array<FilmData>;
+    sortData: SortData;
+    filterValue: string;
+    searchValue: string;
+    currentFilmDisplayed: FilmData | undefined;
+}
+
+const DEFAULT_SORT = {
+    title: 'Release date',
+    direction: 'asc',
+    value: 'release_date',
+};
+
+const INITIAL_STATE: DataSliceInterface = {
+    loading: false,
+    error: '',
+    data: [],
+    sortData: DEFAULT_SORT,
+    filterValue: 'All',
+    searchValue: '',
+    currentFilmDisplayed: undefined,
+};
+
+export const fetchSortedFilteredSearchedMovies = createAsyncThunk<
+    Array<FilmData>,
+    {
+        searchValue: string;
+        filterValue: string;
+        sortData: SortData;
+    }
+>('data/fetchCustomData', async (args) => {
+    const { searchValue, filterValue, sortData } = args;
+    let url = 'http://localhost:4000/movies?';
+
+    url += searchValue ? `search=${searchValue}&searchBy=title&` : '';
+    url +=
+        filterValue && filterValue.toLowerCase() !== 'all'
+            ? `filter=${filterValue}&`
+            : '';
+    url += sortData
+        ? `sortBy=${sortData.value}&sortOrder=${sortData.direction}&`
+        : '';
+    url += 'limit=200';
+
+    const response = await fetchMovies(encodeURI(url));
+    return response.data;
+});
+
+const fetchUpdatedData = (state: any, dispatch: any) => {
+    const { searchValue, filterValue, sortData } = state;
+
+    dispatch(
+        fetchSortedFilteredSearchedMovies({
+            searchValue,
+            filterValue,
+            sortData,
+        })
+    );
+};
+
+export const deleteMovie = createAsyncThunk<void, number, { state: any }>(
+    'data/deleteMovie',
+    async (id, { getState, dispatch }) => {
+        const url = `http://localhost:4000/movies/${id}`;
```

Comment:
the same https://github.com/vmashnitskaya/react-mentoring-program-tasks/pull/5/files#r603335924